### PR TITLE
Add linting to CI / Testing

### LINKED ISSUE #117: Adhere to linting + add linting to CI
Should linting be adding to CI to assure specific rules such as quotations, white space, etc etc etc are followed?

### PR DESCRIPTION
**Pull request by TheAlphaNerd**
This patch closes #117 

It adds an npm script to lint the front end code in ```public/js```.  It also adds the linting step to ```npm test```.  Now linting will be part of the travis CI process, and all code will be nice and tidy.

### CODE REVIEW

File: lib/profile.js [hunk 1]
```diff
@@ -289,7 +289,7 @@ exports.exportPosts = function (request, reply) {
       return reply(err);
     }
 
-    var posts = all.posts.map(function (post) {
```
Comment (TheAlphaNerd) [inline lib/profile.js]: posts is already declared and being shadowed here

File: lib/twilio.js [hunk 2]
```diff
@@ -5,25 +6,23 @@ var mock = function () {
...
-}
+};
+
 var isTest = process.env.NODE_ENV === 'test';
 var isDev = process.env.npm_lifecycle_event === 'dev';
 
 if (isTest || isDev) {
-  
   module.exports = mock;
-  return;
```
Comment (TheAlphaNerd) [inline lib/twilio.js]: the illegal return was throwing in the linter

Comment (jasonrhodes) [inline lib/twilio.js]: Yeah we either need to change that linting rule or wrap the rest of the file in an else block or the work in that Twilio mock gets totally undone.

Comment (jasonrhodes) [inline lib/twilio.js]: Ah I didn't see the added else block at first. Sounds good.

**Comment by jasonrhodes** [comment]
@TheAlphaNerd this looks good, except can we turn off the `no-unused-vars` rule everywhere? I'm getting a lot of those warnings when I run the lints, and that error is pretty useless IMO...

**Comment by TheAlphaNerd** [comment]
We can turn off no-unused-vars... although I left it in as a warn for a reason.

It is getting some false positives in the case of functions that have declared parameters that are unused, although perhaps those should arguably be removed (or at least commented out).  There are some instances where it flags on variables that legitimately should not be there such as the inclusion of rimraf in ```lib/db.js```

So we are left with the choice of disabling it and letting unused variables in the code... or making a decision about what to do with the unused vars in functions.

**Comment by jasonrhodes** [comment]
That makes sense. Warns don't fail Travis, right?

**Comment by TheAlphaNerd** [comment]
Nope.  I'll submit a seperate PR that will take a stab at getting rid of the warnings.  Once we get rid of them we can turn warn off.

**Comment by jasonrhodes** [comment]
Sounds good, let's get lintin'...

**Pull request merged**